src/js/control-bar/text-track-controls/tracks-button.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file tracks-button.js
+ */
+import TextTrackButton from './text-track-button.js';
+import Component from '../../component.js';
+import CaptionSettingsMenuItem from './caption-settings-menu-item.js';
+
+/**
+ * The button component for toggling and selecting captions
+ *
+ * @extends TextTrackButton
+ */
+class TracksButton extends TextTrackButton {
+
+  /**
+   * Creates an instance of this class.
+   *
+   * @param {Player} player
+   *        The `Player` that this class should be attached to.
+   *
+   * @param {Object} [options]
+   *        The key/value store of player options.
+   *
+   * @param {Component~ReadyCallback} [ready]
+   *        The function to call when this component is ready.
+   */
+  constructor(player, options, ready) {
+    super(player, options, ready);
+    this.el_.setAttribute('aria-label', 'Captions Menu');
+  }
+
+  /**
+   * Builds the default DOM `className`.
+   *
+   * @return {string}
+   *         The DOM `className` for this object.
+   */
+  buildCSSClass() {
+    return `vjs-tracks-button ${super.buildCSSClass()}`;
+  }
+
+  /**
+   * Update caption menu items
+   *
+   * @param {EventTarget~Event} [event]
+   *        The `addtrack` or `removetrack` event that caused this function to be
+   *        called.
+   *
+   * @listens TextTrackList#addtrack
+   * @listens TextTrackList#removetrack
+   */
+  update(event) {
+    let threshold = 2;
+
+    super.update();
+
+    // if native, then threshold is 1 because no settings button
+    if (this.player().tech_ && this.player().tech_.featuresNativeTextTracks) {
+      threshold = 1;
+    }
+
+    if (this.items && this.items.length > threshold) {
+      this.show();
+    } else {
+      this.hide();
+    }
+  }
+
+  /**
+   * Create caption menu items
+   *
+   * @return {CaptionSettingsMenuItem[]}
+   *         The array of current menu items.
+   */
+  createItems() {
+    const items = [];
+
+    if (!(this.player().tech_ && this.player().tech_.featuresNativeTextTracks)) {
+      // TODO - kind?
+      items.push(new CaptionSettingsMenuItem(this.player_, {kind: 'captions'}));
+    }
+
+    return super.createItems(items);
+  }
+
+}
+
+/**
+ * `kind`s of TextTrack to look for to associate it with this menu.
+ *
+ * @type {array}
+ * @private
+ */
+TracksButton.prototype.kinds_ = ['captions', 'subtitles'];
+
+/**
+ * The text that should display over the `TracksButton`s controls. Added for localization.
+ *
+ * @type {string}
+ * @private
+ */
+TracksButton.prototype.controlText_ = 'Captions';
```

Comment:
And a little more complex to localise if the string needs to be localised as captions or subtitles here, but specifically as captions on the captions-only control, which might be translated as "subtitles for the hard of hearing".

This would be easier if we used something like CAPTIONS_OR_SUBTITLES as keys in the translations json and included en.json in the player.